bindings/rust/wasmedge-sys/src/vm.rs

```diff
@@ -166,11 +193,143 @@ impl Vm {
         }
         Ok(returns.into_iter().map(Into::into).collect())
     }
+
+    pub fn statistics(&self) -> Statistics {
+        let ctx = unsafe { wasmedge::WasmEdge_VMGetStatisticsContext(self.ctx) };
+        Statistics {
+            ctx,
+            registered: true,
+        }
+    }
 }
 
 impl Drop for Vm {
     fn drop(&mut self) {
-        unsafe { wasmedge::WasmEdge_VMDelete(self.ctx) };
-        self.import_objects.drain(..);
+        if !self.ctx.is_null() {
+            unsafe { wasmedge::WasmEdge_VMDelete(self.ctx) };
+        }
+        if !self.import_objects.is_empty() {
+            self.import_objects.drain(..);
```

Comment:
Is there any issue to drain an empty vector?
Or is there any beneficial to drain a vector before checking it is empty or not?

Reply:
As for the drain method, it can work with an empty Vec instance. The import_objects field is introduced in the original code. To avoid destroying the original design logic, keep it there in this PR. It would be better to decide to either remove it or refactor it in the future. Thanks!

Reply:
Will remove the redundant check code in line 211, thanks!